### PR TITLE
Accept more technically correct captcha inputs

### PR DESCRIPTION
**Pull request by Philimanjaroo**
intented new functionalities:

1) transform input into time formats from 00:00-11:59.

2) add "0" if not given (e.g. 3:30 becomes 03:30).

### CODE REVIEW

File: src/api/main/SignupForm.js [hunk 1]
```diff
@@ -203,7 +203,19 @@ function runCaptcha(mailAddress: string, isBusinessUse: boolean, isPaidSubscript
...
+								captchaTime = '0' + captchaTime;
+							} 
+							if ((Number(captchaTime.substr(0,2))) >= 12 && (Number(captchaTime.substr(0,2))) < 22) {
+								let hours = Number(captchaTime.substr(0,2))-12;
+								captchaTime = "0"+ hours + captchaTime.substr(2,string_length);
+								} else if (Number(captchaTime.substr(0,2)) > 21 && Number(captchaTime.substr(0,2)) <= 23) {
+									let hours = Number(captchaTime.substr(0,2))-12;
+									captchaTime = hours + captchaTime.substr(2,string_length);
+								} else if (Number(captchaTime.substr(0,2)) < 12) {}
+
```
Comment (jowlo) [inline src/api/main/SignupForm.js:218]: Thank you!

But this code is quite hard to read and i think a bit too complex. I had a quick take at it that i find easier to read and it does less string manipulations. Feel free to comment

```javascript
// Make numbers from both tokens
[h, m] = captchaTime.split(':').map(t => Number(t))
// Cut numbers to hour and minute format, left pad with zeros, and glue with a colon
captchaTime = [h % 12, m % 60].map(a => String(a).padStart(2,'0')).join(':')
```

Comment (Philimanjaroo) [reply src/api/main/SignupForm.js:218]: That's a concise solution. I like it very much. :+1:

Comment (jowlo) [reply src/api/main/SignupForm.js:218]: Cool, i made a pull request with my variant.
But still: Thank you for your contribution! :+1:

**Comment by charlag** [comment]
Superseded by #2607

**Pull request closed without merging**